### PR TITLE
Add support for extended SANs on certificate requests

### PR DESCRIPTION
**Pull request by maraino**
### Description

This commit supports extended SANs like the permanent identifier used in device attestation flows with CSRs. Step already has support for this on certificates but not on certificate requests.

With this, it will be possible to run:
```shell
step certificate create --csr --template att.tpl \
--kms "yubikey:?pin-value=123456" --key "yubikey:slot-id=82" \
123456789 att.csr
```

Where att.tpl can be something like:
```tpl
{
  "subject": {{ toJson .Subject }},
  "sans": [{
    "type": "permanentIdentifier",
    "value": {{ toJson .Subject.CommonName }}
  }]
}
```

### CODE REVIEW

File: x509util/extensions_test.go [hunk 1]
```diff
@@ -1280,33 +1303,50 @@ func Test_createSubjectAltNameExtension(t *testing.T) {
...
-			got, err := createSubjectAltNameExtension(tt.args.c, tt.args.subjectIsEmpty)
+			gotCert, err := createCertificateSubjectAltNameExtension(tt.args.c, tt.args.subjectIsEmpty)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("createSubjectAltNameExtension() error = %v, wantErr %v", err, tt.wantErr)
 				return
 			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("createSubjectAltNameExtension() = \n%v, want \n%v", got, tt.want)
+			if !reflect.DeepEqual(gotCert, tt.want) {
+				t.Errorf("createSubjectAltNameExtension() = \n%v, want \n%v", gotCert, tt.want)
```
Comment (hslatman) [inline x509util/extensions_test.go:1332]:
```suggestion
				t.Errorf("createCertificateSubjectAltNameExtension() = \n%v, want \n%v", gotCert, tt.want)
```

File: x509util/extensions_test.go [hunk 2]
```diff
@@ -1280,33 +1303,50 @@ func Test_createSubjectAltNameExtension(t *testing.T) {
...
+				DNSNames:       tt.args.c.DNSNames,
+				EmailAddresses: tt.args.c.EmailAddresses,
+				IPAddresses:    tt.args.c.IPAddresses,
+				URIs:           tt.args.c.URIs,
+				SANs:           tt.args.c.SANs,
+			}
+
+			gotCSR, err := createCertificateRequestSubjectAltNameExtension(cr, tt.args.subjectIsEmpty)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("createSubjectAltNameExtension() error = %v, wantErr %v", err, tt.wantErr)
```
Comment (hslatman) [inline x509util/extensions_test.go:1345]:
```suggestion
				t.Errorf("createCertificateRequestSubjectAltNameExtension() error = %v, wantErr %v", err, tt.wantErr)
```

File: x509util/extensions_test.go [hunk 3]
```diff
@@ -1280,33 +1303,50 @@ func Test_createSubjectAltNameExtension(t *testing.T) {
...
+				return
+			}
+			if !reflect.DeepEqual(gotCSR, tt.want) {
+				t.Errorf("createSubjectAltNameExtension() = \n%v, want \n%v", gotCSR, tt.want)
```
Comment (hslatman) [inline x509util/extensions_test.go:1349]:
```suggestion
				t.Errorf("createCertificateRequestSubjectAltNameExtension() = \n%v, want \n%v", gotCSR, tt.want)
```

File: x509util/extensions_test.go [hunk 4]
```diff
@@ -1280,33 +1303,50 @@ func Test_createSubjectAltNameExtension(t *testing.T) {
...
 				{Type: "1.2.3.4", Value: "int:bar@foo.com"},
 			},
 		}, false}, Extension{}, true},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := createSubjectAltNameExtension(tt.args.c, tt.args.subjectIsEmpty)
+			gotCert, err := createCertificateSubjectAltNameExtension(tt.args.c, tt.args.subjectIsEmpty)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("createSubjectAltNameExtension() error = %v, wantErr %v", err, tt.wantErr)
```
Comment (hslatman) [inline x509util/extensions_test.go:1328]:
```suggestion
				t.Errorf("createCertificateSubjectAltNameExtension() error = %v, wantErr %v", err, tt.wantErr)
```

**Review by hslatman — COMMENTED** [review]
Minor things to replace

File: x509util/extensions_test.go (hunk 1 above)
Comment (maraino) [reply x509util/extensions_test.go]: Fixed with fafb878

File: x509util/extensions_test.go (hunk 2 above)
Comment (maraino) [reply x509util/extensions_test.go]: Fixed with fafb878

File: x509util/extensions_test.go (hunk 3 above)
Comment (maraino) [reply x509util/extensions_test.go]: Fixed with fafb878

File: x509util/extensions_test.go (hunk 4 above)
Comment (maraino) [reply x509util/extensions_test.go]: Fixed with fafb878

**Review by hslatman — APPROVED** [review]

**Pull request merged**